Docs/source/running_cpp/parameters.rst

```diff
@@ -1741,3 +1741,101 @@ This is essentially the python slicing syntax except that the stop is inclusive
+    If set to ``parse_H_bias_ext_grid_function``, then a mathematical expression can
+    be used to initialize the magnetic bias on the grid. It
+    requires additional parameters in the input file, namely,
+    ``warpx.Hx_bias_external_grid_function(x,y,z)``,
+    ``warpx.Hy_bias_external_grid_function(x,y,z)``,
+    ``warpx.Hz_bias_external_grid_function(x,y,z)`` to initialize the external
+    magnetic bias for each of the three components on the grid.
+    Constants required in the expression can be set using ``my_constants``.
+    For example, if ``warpx.Hx_bias_external_grid_function(x,y,z)=Ho_bias*x + delta*(y + z)``
+    then the constants `Ho_bias` and `delta` required in the above equation
+    can be set using ``my_constants.Ho_bias=`` and ``my_constants.delta=`` in the
+    input file. For a two-dimensional simulation, it is assumed that the first dimension     is `x` and the second dimension in `z`, and the value of `y` is set to zero.
+    Note that the current implementation of the parser for H_bias-field
+    does not work with RZ and the code will abort with an error message.
+    If ``algo.em_solver_medium`` is set to macroscopic, and ``USE_LLG = TRUE``, 
+    then this input parameter must be provided.
+
+* ``warpx.M_external_grid`` & ``warpx.H_bias_external_grid`` (list of `double`)
+    required when ``warpx.M_ext_grid_init_style="constant"``
+    and when ``warpx.H_bias_ext_grid_init_style="constant"``, respectively.
+    External uniform and constant magnetization and magnetostatic bias field added
+    to the grid at initialization. Use with caution as these fields are used for
+    the field solver. In particular, do not use any other boundary condition
+    than periodic.
+
+* ``macroscopic.mag_Ms_init_style`` (string) optional (default is "default")
+    This parameter determines the type of initialization for the saturation magnetization
+    of the material. The "default" style initializes the saturation magnetization mag_Ms to 0.0.
+    The string can be set to "constant" if a constant saturation magnetization is
+    required to be set at initialization. If set to "constant", then an
+    additional parameter, namely, ``macroscopic.mag_Ms`` must be specified.
+    If set to ``parse_mag_Ms_function``, then a mathematical expression can
+    be used to initialize the saturation magnetization on the grid. It
+    requires additional parameters in the input file, namely,
+    ``macroscopic.mag_Ms_function(x,y,z)`` to initialize the saturation magnetization.
+    If ``algo.em_solver_medium`` is set to macroscopic, and ``USE_LLG = TRUE``, 
+    then this input property must be provided.
+
+* ``macroscopic.mag_alpha_init_style`` (string) optional (default is "default")
+    This parameter determines the type of initialization for the Gilbert damping factor
+    of the material. The "default" style initializes the Gilbert damping factor mag_alpha to 0.0.
```

Comment:
Note that only positive (or zero) values for Gilbert damping factor are allowed in the simulation.